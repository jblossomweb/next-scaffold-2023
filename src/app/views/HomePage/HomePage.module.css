.container {
  @apply px-8 py-0;
}

.main {
  @apply min-h-screen
    px-0 py-16
    flex flex-col
    justify-center
    items-center;
}

.title a {
  @apply text-blue-500
    no-underline;
}

.title a:hover,
.title a:focus,
.title a:active {
  @apply underline;
}

.title {
  @apply m-0
    leading-5
    text-6xl;
}

.title,
.description {
  @apply text-center;
}

.description {
  @apply mx-0 my-16
    leading-5
    text-base;
}

.code {
  @apply p-2 rounded-md
    bg-slate-300
    text-sm
    font-mono;
}

.footer {
  @apply mx-0 my-0;
}
