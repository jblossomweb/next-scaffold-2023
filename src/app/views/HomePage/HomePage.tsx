import React from 'react'
import type { NextPage } from 'next'
import Head from 'next/head'

import Image from 'app/components/atoms/Image'
import styles from './HomePage.module.css'

const HomePage: NextPage = () => (
  <div className={styles.container}>
    <Head>
      <title>Create Next App</title>
      <meta name="description" content="Generated by create next app" />
      <link rel="icon" href="/favicon.ico" />
    </Head>

    <main className={styles.main}>
      <h1 className={styles.title}>
        Welcome to <a href="https://nextjs.org">Next.js!</a>
      </h1>

      <p className={styles.description}>
        Get started by editing{' '}
        <code className={styles.code}>app/views/HomePage/HomePage.tsx</code>
      </p>
      <div className={styles.footer}>
        <Image alt="Next.js" src="/nextjs.png" width={80} height={48} />
      </div>
    </main>
  </div>
)

export default HomePage
