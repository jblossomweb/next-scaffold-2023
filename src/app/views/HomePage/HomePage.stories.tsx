import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import HomePage from './HomePage'

export default {
  title: 'views/HomePage',
  component: HomePage,
} as Meta<typeof HomePage>

const Template: StoryFn<typeof HomePage> = () => <HomePage />

export const Sample = Template.bind({})
