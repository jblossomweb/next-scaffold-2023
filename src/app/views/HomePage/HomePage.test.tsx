/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import Index from './index'
import * as stories from './HomePage.stories'

describe('app/views/HomePage', () => {
  describe('index', () => {
    afterEach(cleanup)

    it('should wireup without crashing', () => {
      render(<Index />)
    })
  })

  describe('component', () => {
    describe('stories', () => {
      storySnaps(stories)
    })
  })
})
