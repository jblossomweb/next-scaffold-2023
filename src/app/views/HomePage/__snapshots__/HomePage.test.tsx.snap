// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/views/HomePage component stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="description"
      >
        Get started by editing 
        <code
          class="code"
        >
          app/views/HomePage/HomePage.tsx
        </code>
      </p>
      <div
        class="footer"
      >
        <figure
          class=""
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2748%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Next.js"
              aria-label="Next.js"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              title="Next.js"
            />
            <noscript />
          </span>
        </figure>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
